### ISSUE TITLE
Vagrant 1.8.6: hangs at step "Configuring and enabling network interfaces..." if private_network is enabled

### ISSUE THREAD

**Issue by amatas** [mitchellh/vagrant#7950]
### Vagrant version
Vagrant 1.8.6

### Host operating system
Archlinux

### Guest operating system
Centos 7 + Docker installed and enabled
For example: https://atlas.hashicorp.com/williamyeh/boxes/centos7-docker

### Vagrantfile
```ruby
# -*- mode: ruby -*-
# vi: set ft=ruby :

Vagrant.configure("2") do |config|
  config.vm.box = "williamyeh/centos7-docker"
  config.vm.network "private_network", ip: "192.168.33.10"
end
```

### Debug output
https://gist.github.com/amatas/10bbf7d691d5443e74f89af42f12ec93#file-vagrant_interface-log-L4248-L4249

### Expected behavior
- The interface enp0s3 should be configured as internal interface **10.0.2.15**
- The interface enp0s8 should be configured as private interface **192.168.33.10**

### Actual behavior
Vagrant is taking the interface **docker0** as internal interface as the log shows, breaking the configuration of the next interface in the list **enp0s3**. As the internal interface is down, Vagrant is not able to contact with the VM causing the "hang".

### Steps to reproduce
1. vagrant init williamyeh/centos7-docker
2. Use the above Vagrantfile
3. vagrant up
4. The provision hangs at "Configuring and enabling network interfaces..." step

**Comment by looppair** [comment]
Seeing the [same behaviour](https://unix.stackexchange.com/q/320283/3645) on a custom CentOS 7 box using VirtualBox & GuestAdditions 5.0.28, Vagrant 1.8.1 and vagrant-vbguest 0.13.0.

**Comment by chrisroberts** [comment]
#7866 should have resolved this issue which is in the 1.8.7 release. Please let me know if it does not resolve your issue. Thanks!

**Comment by amatas** [comment]
The issue is fixed in 1.8.7 release using the same environment described above. Thanks.

**Comment by chrisroberts** [comment]
:+1: thanks for the update
